### PR TITLE
Fix docs

### PR DESCRIPTION
**Pull request by jaimergp**
After #680 

# To do

- [ ] Add CI to make sure docs build without warnings.
- [ ] Ensure mocking works instead of installing all dependencies. Some are too complex to mock, but at least we will get rid of most.
- [ ] Make sure these changes apply to readthedocs as well

### CODE REVIEW

**Comment by jaimergp** [comment]
Check [logs for docs](https://github.com/choderalab/perses/pull/682/checks?check_run_id=730851873#step:5:10277) to see all warnings. It's not as broken as before, but there's plenty of pointers to dead / moved code and I am not very familiar with those. It might be also helpful to disable `numba`'s ultra verbose DEBUG logging.

**Comment by jaimergp** [comment]
Also, imports and the logging machinery needs some work. Not in this PR, but just writing it down for future reference.

File: environment.yml [hunk 1]
```diff
@@ -32,3 +32,5 @@ dependencies:
   - tqdm
   - nose
   - nose-timer
+  - openforcefield
+  - sphinx
```
Comment (dominicrufa) [inline environment.yml:36]: this enable documentation?

File: docs/rjmc.rst [hunk 2]
```diff
@@ -15,16 +15,17 @@ Topology proposal engines
...
     PeptideLibraryEngine
-    NullProposalEngine
-    NaphthaleneProposalEngine
-    ButaneProposalEngine
-    PropaneProposalEngine
+    .. JRG: This class does not exist anymore?
+    .. ButaneProposalEngine
+    .. NaphthaleneProposalEngine
+    .. PropaneProposalEngine
+    .. NullProposalEngine
```
Comment (dominicrufa) [inline docs/rjmc.rst:28]: this allows for documentation of classes? this should actually be for `topology_proposal.py`, no?

File: docs/rjmc.rst [hunk 3]
```diff
@@ -34,7 +35,8 @@ OpenMM System generation utilities
     :nosignatures:
     :toctree: api/generated/
 
-    SystemGenerator
+    .. .. This has moved to openforcefields?
```
Comment (dominicrufa) [inline docs/rjmc.rst:38]: yes

File: docs/rjmc.rst [hunk 4]
```diff
@@ -46,7 +48,8 @@ Geometry proposal engines
 
     GeometryEngine
     FFAllAngleGeometryEngine
-    OmegaGeometryEngine
+    .. JRG: This class does not exist anymore?
+    .. OmegaGeometryEngine
```
Comment (dominicrufa) [inline docs/rjmc.rst:52]: this was removed

File: docs/rjmc.rst [hunk 5]
```diff
@@ -56,9 +59,11 @@ Geometry utility classes
...
-    GeometrySystemGeneratorFast
-    PredHBond
-    ProposalOrderTools
+    .. JRG: These classes does not exist anymore?
+    .. BootstrapParticleFilter
+    .. GeometrySystemGeneratorFast
+    .. PredAtomTopologyIndex
+    .. PredHBond
+    .. .. This one is also dead code in some modules!
+    .. ProposalOrderTools
```
Comment (dominicrufa) [inline docs/rjmc.rst:69]: i mostly rewrote `geometry.py` and never updated documentation. should there be a .rst for every API-containing `.py` script in the code?

Comment (jaimergp) [reply docs/rjmc.rst:69]: That would be one way. The current approach is a semi-automated way to approach this, but it should be rearranged into a different directory tree. I like the `api` stuff to be separate from the user-oriented docs. I can propose a new scheme, but first we need to trim out all the dead leaves :)

If you can add comments to those classes that are gone in the current version and identify if the missing references (check the logs on `CI / Build documentation`) are a bug or just the byproduct of dead code, then I can work on the rest! Thanks!

**Comment by hannahbrucemacdonald** [comment]
@jaimergp @dominicrufa @jchodera could this be merged soon? It is only changes to the docs, and while the changes aren't complete, they are still an improvement and the longer it's left the more conflicts there will be to merge

**Pull request merged**